### ISSUE TITLE
Cache hit rate

### ISSUE THREAD

**Issue by daviswr** [daviswr/ZenPacks.daviswr.OSX.Server.Caching#7]
```
$ grep "Since server start" /Library/Server/Caching/Logs/Service.log | tail -1
2018-07-04 08:54:06.022 Since server start: 5.73 GB returned to clients, 5.45 GB stored from Internet, 0 bytes from peers; 0 bytes imported.
```

**Comment by daviswr** [comment]
```
$ sqlite3 /Library/Server/Caching/Logs/Metrics.sqlite "SELECT metricName, dataValue FROM statsData WHERE collectionDate = (SELECT Max(collectionDate) FROM statsData)"
bytes.fromcache.topeers|0
requests.fromclients|1
imports.byhttp|0
bytes.frompeers.toclients|0
bytes.purged.total|0
replies.fromorigin.topeers|0
bytes.purged.youngerthan1day|0
bytes.fromcache.toclients|0
bytes.imported.byxpc|0
requests.frompeers|0
bytes.fromorigin.toclients|24307464
replies.fromcache.topeers|0
bytes.imported.byhttp|0
bytes.dropped|0
replies.fromcache.toclients|0
replies.frompeers.toclients|0
imports.byxpc|0
bytes.purged.youngerthan7days|0
bytes.fromorigin.topeers|0
replies.fromorigin.toclients|1
bytes.purged.youngerthan30days|0
```

**Comment by daviswr** [comment]
Stats seem to be generated every two days

```
$ sqlite3 /Library/Server/Caching/Logs/Metrics.sqlite "SELECT DISTINCT Date(collectionDate, 'unixepoch') FROM statsData"
2018-06-30
2018-07-02
2018-07-04
```

**Comment by daviswr** [comment]
Already discussed in detail at https://themacwrangler.wordpress.com/2017/02/21/monitoring-apple-caching-server/

**Comment by daviswr** [comment]
High Sierra is similar... ish

```
$ sqlite3 --line "/Library/Application Support/Apple/AssetCache/Metrics/Metrics.db" "SELECT * FROM ZMETRIC"
                         Z_PK = 1
                        Z_ENT = 3
                        Z_OPT = 1
                ZBYTESDROPPED = 0
       ZBYTESFROMCACHETOCHILD = 0
      ZBYTESFROMCACHETOCLIENT = 0
        ZBYTESFROMCACHETOPEER = 0
      ZBYTESFROMORIGINTOCHILD = 0
     ZBYTESFROMORIGINTOCLIENT = 0
       ZBYTESFROMORIGINTOPEER = 0
      ZBYTESFROMPARENTTOCHILD = 0
     ZBYTESFROMPARENTTOCLIENT = 0
       ZBYTESFROMPARENTTOPEER = 0
        ZBYTESFROMPEERTOCHILD = 0
       ZBYTESFROMPEERTOCLIENT = 9836637
         ZBYTESIMPORTEDBYHTTP = 0
          ZBYTESIMPORTEDBYXPC = 0
            ZBYTESPURGEDTOTAL = 0
  ZBYTESPURGEDYOUNGERTHAN1DAY = 0
ZBYTESPURGEDYOUNGERTHAN30DAYS = 0
 ZBYTESPURGEDYOUNGERTHAN7DAYS = 0
               ZIMPORTSBYHTTP = 0
                ZIMPORTSBYXPC = 0
                      ZPERIOD = 60
     ZREPLIESFROMCACHETOCHILD = 0
    ZREPLIESFROMCACHETOCLIENT = 0
      ZREPLIESFROMCACHETOPEER = 0
    ZREPLIESFROMORIGINTOCHILD = 0
   ZREPLIESFROMORIGINTOCLIENT = 0
     ZREPLIESFROMORIGINTOPEER = 0
    ZREPLIESFROMPARENTTOCHILD = 0
   ZREPLIESFROMPARENTTOCLIENT = 0
     ZREPLIESFROMPARENTTOPEER = 0
      ZREPLIESFROMPEERTOCHILD = 0
     ZREPLIESFROMPEERTOCLIENT = 3
           ZREQUESTSFROMCHILD = 0
          ZREQUESTSFROMCLIENT = 3
            ZREQUESTSFROMPEER = 0
  ZREQUESTSREJECTEDFORNOSPACE = 0
                ZCREATIONDATE = 553299761.112068

$ sqlite3 --line "/Library/Application Support/Apple/AssetCache/Metrics/Metrics.db" "SELECT DISTINCT Date(ZCREATIONDATE, 'unixepoch') FROM ZMETRIC"
Date(ZCREATIONDATE, 'unixepoch') = 1987-07-14
```

**Comment by daviswr** [comment]
`ZCREATIONDATE` appears to be seconds since 1 Jan 2001, in UTC - [https://www.epochconverter.com/coredata](https://www.epochconverter.com/coredata)

**Comment by daviswr** [comment]
```
sqlite3 "/Library/Application Support/Apple/AssetCache/Metrics/Metrics.db" "SELECT DISTINCT DateTime(978307200 + ZCREATIONDATE, 'unixepoch') FROM ZMETRIC"
2021-03-12 16:43:00
2021-03-12 17:37:00
2021-03-12 17:38:00
2021-03-12 17:39:00
2021-03-12 20:47:00
```

**Comment by daviswr** [comment]
Closed by 124c636 and a80a0dc
